tests/spark/test_spark_model_export.py

```diff
@@ -77,23 +117,42 @@ def test_model_log(tmpdir):
     preds_df = model.transform(spark_df)
     preds1 = [x.prediction for x in preds_df.select("prediction").collect()]
     old_tracking_uri = tracking.get_tracking_uri()
+    cnt = 0
     # should_start_run tests whether or not calling log_model() automatically starts a run.
     for should_start_run in [False, True]:
-        try:
-            tracking_dir = os.path.abspath(str(tmpdir.mkdir("mlruns")))
-            tracking.set_tracking_uri("file://%s" % tracking_dir)
-            if should_start_run:
-                tracking.start_run()
-            sparkm.log_model(artifact_path="model", spark_model=model)
-            run_id = tracking.active_run().info.run_uuid
-            x = pyfunc.load_pyfunc("model", run_id=run_id)
-            preds2 = x.predict(pandas_df)
-            assert preds1 == preds2
-            reloaded_model = sparkm.load_model("model", run_id=run_id)
-            preds_df_1 = reloaded_model.transform(spark_df)
-            preds3 = [x.prediction for x in preds_df_1.select("prediction").collect()]
-            assert preds1 == preds3
-        finally:
-            tracking.end_run()
-            tracking.set_tracking_uri(old_tracking_uri)
-            shutil.rmtree(tracking_dir)
+        for dfs_tmp_dir in [None, os.path.join(tmpdir, "test")]:
```

Comment:
Could we add a log statement indicating which iteration we're in, to help diagnosis if it fails?